tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientTimeoutTest.java

```diff
@@ -520,7 +521,7 @@ private void assumeConnectTimeout(String host, int port, int connectTimeout)
         catch (Throwable x)
         {
             // Abort if any other exception happens.
-            fail(x);
+            throw new TestAbortedException("Not able to validate connect timeout conditions", x);
```

Comment:
Please use `Assumptions` instead of throwing directly - it's not clear if you want the test to fail or what.

Reply:
The test will abort, not fail.

Using `Assumptions` makes the test either continue or abort, never fail.
Abort marks the test as ignored.

Using `TestAbortedException` preserves the stacktrace, which is then reported when a test is aborted. (this information is even present in your IDE)
Using `Assumptions` loses this stacktrace.

Reply:
We do a similar thing in the AllowSymLinkAliasCheckerTest.

https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-server/src/test/java/org/eclipse/jetty/server/handler/AllowSymLinkAliasCheckerTest.java

Reply:
Bleah!